test/models/position_test.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/util/search/position.dart';
+
+void main() {
+  group('Named constructors', () {
+    test('Position.fromJson', () {
+      final Map<String, dynamic> json = {
+        //TODO Is is possible for this to be an int?
+        'lat': 50.0,
+        'lng': 8.2714,
+      };
+      final Position position = Position.fromJson(json);
+      expect(position.lat, 50);
+      expect(position.lng, 8.2714);
+    });
+
+    test('Position.fromDynamicValues', () {
+      final Map<String, dynamic> json = {
+        'lat': 50,
+        'lng': 8.2714,
+      };
+      final Position position = Position.fromDynamicValues(json['lat'], json['lng']);
+      expect(position.lat, 50);
+      expect(position.lng, 8.2714);
+    });
+  });
+
+  group('ProfileFeature.toJson', () {
+    test('returns a json representation of the ProfileFeature', () async {
+      final Position position = Position(50, 8.2714);
+      final Map<String, dynamic> json = position.toJson();
+      expect(json['lat'], 50);
+      expect(json['lng'], 8.2714);
+      expect(json.keys.length, 2);
+    });
+  });
+
+  group('Position ==', () {
+    test('equal', () {
+      final Position position1 = Position(50, 8.2714);
+      final Position position2 = Position(50, 8.2714);
+      expect(position1 == position2, isTrue);
+    });
+
+    test('not equal', () {
+      final Position position1 = Position(50, 8.2714);
+      final Position position2 = Position(50, 8.2715);
+      expect(position1 == position2, isFalse);
+    });
+  });
+
+  group('Position.hashCode', () {
+    test('Mainz', () {
+      final Position position = Position(50, 8.2714);
+      expect(position.hashCode, 9159982799919094);
```

Comment:
I had to change this value just now because I think the new dart version just returned something different. That reminds me that I actually hate this test, does anyone have a better idea how to test this or why we even have the method `Position.hashCode`?

Reply:
You need to implement `hashCode` when you implement the `==` operator. Maybe that stupid fact alone would be reason to use this package instead: https://pub.dev/packages/equatable (Which generates those for you)

Reply:
I think you just need to make sure that hashCode is equal if `==` considers the both equal (keep in mind, for some reason not necessarily _exactly if_, just if). So maybe just test that fact?

Reply:
So
```
final Position position1 = Position(50, 8.2714);
final Position position2 = Position(50, 8.2714);
expect(position1.hashCode, position2.hashCode);
```
? Boy do I not like this test

Reply:
Yes that seems like a better test. And then

```
final Position otherPosition = Position(50, 8.2715);
expect(position1.hashCode == position2.hashCode, false);
```